src/ir/possible-contents.cpp

```diff
@@ -988,7 +988,8 @@ struct InfoCollector
     auto* set = builder.makeArraySet(curr->destRef, curr->destIndex, get);
     visitArraySet(set);
   }
-
+  void visitArrayFill(ArrayFill* curr) { WASM_UNREACHABLE("unimplemented"); }
+  void visitArrayInit(ArrayInit* curr) { WASM_UNREACHABLE("uimplemented"); }
```

Comment:
The fuzzer will crash on these (any testcase with the instructions + `--gufa`).

I think their implementation can be similar to `ArraySet`.

Reply:
What should I do to model reading from the data or element segment in `ArrayInit`?

Reply:
Looks like `TableGet` just models the value as a root (an unknown value that could be anything). I think we should do the same here. But that value would need to be written into the array's storage instead of just declaring this expression to be a root.

If it doesn't look obvious how to do that I can look into it. I'm not sure if it will be trivial or not.

Reply:
Thanks. I'm not sure how to hook up the root value with the synthetic write, so I'll leave that to you and look forward to seeing what's involved.

Reply:
My fuzzing did find this after 7591 iterations, so at least it's rare.

Reply:
Ah, I was hoping to fix this before it lands, as that frequency still means it'll be found many times a day. But no worries, I'll look into this right now.

Reply:
Oops, sorry about that.

Reply:
No problem, I think I found the solution already, PR is up (#5638)